CMakeLists.txt

```
@@ -857,8 +857,8 @@ if(WITH_OPTIM)
             check_sse42_intrinsics()
             if(HAVE_SSE42_INTRIN AND WITH_SSSE3)
                 add_definitions(-DX86_SSE42)
-                set(SSE42_SRCS ${ARCHDIR}/adler32_sse42.c ${ARCHDIR}/insert_string_sse42.c)
-                add_feature_info(SSE42_CRC 1 "Support SSE4.2 optimized CRC hash generation, using \"${SSE42FLAG}\"")
+                set(SSE42_SRCS ${ARCHDIR}/adler32_sse42.c)
+                add_feature_info(SSE42_CRC 1 "Support SSE4.2 optimized adler32 hash generation, using \"${SSE42FLAG}\"")
```

```suggestion
                add_feature_info(SSE42_ADLER32 1 "Support SSE4.2 optimized adler32 hash generation, using \"${SSE42FLAG}\"")
```

The check_sse42_intrinsics function is also testing for functions that aren't used anymore.

---

Do you mean that `check_sse42_intrinsics` compiles code containing `return _mm_crc32_u32(a, b);`?
That is not a problem, it doesn't matter what SSE4.2 function we test with since that means they should all work.

With ARM however, we could possibly remove the `+crc` from march specifications. But I feel someone more familiar with that should probably make those changes.